auth_oidc/controllers/main.py

```diff
@@ -48,3 +53,26 @@ def list_providers(self):
                     provider["auth_endpoint"], url_encode(params)
                 )
         return providers
+
+
+class OpenIDLogout(Session):
+    @http.route("/web/session/logout", type="http", auth="none")
+    def logout(self, redirect="/web/login"):
+        # https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RPLogout
+        user = request.env["res.users"].sudo().browse(request.session.uid)
+        if user.oauth_provider_id.id:
+            provider = (
+                request.env["auth.oauth.provider"]
+                .sudo()
+                .browse(user.oauth_provider_id.id)
+            )
+            if provider.end_session_endpoint:
+                red_url = urljoin(request.httprequest.url_root, redirect)
```

Comment:
When I see `red_url` my brain wonders why that url is not blue ;)
Let's not abuse abbreviations.